staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation.go

```diff
@@ -210,6 +210,10 @@ func (r ratchetingOptions) shouldRatchetError() bool {
 func (r ratchetingOptions) key(field string) ratchetingOptions {
 	if r.currentCorrelation == nil {
 		return r
+	} else if r.nearestParentCorrelation == nil && (field == "apiVersion" || field == "kind") {
```

Comment:
Apologies if this is obvious to everyone else, but does `r.nearestParentCorrelation == nil` always guarantee we're at the root of the schema?

Reply:
Sounds good.  I figured you'd had this worked out.

Would a test that shows a nested field that happens to be named "apiVersion" be easy to throw in somewhere? (and be useful)?  Making sure this doesn't get matched is what triggered my interest.

Reply:
Ive added a test